### PR TITLE
Swiss whoami

### PR DESCRIPTION
**Pull request by vascocosta**
Let me know if this is of any interest and if you'd like me to sign the CA.

### CODE REVIEW

File: apps/swiss/whoami.c [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+{
+
+    PSTR UserName;
+
+    UserName = NULL;
+    if (SwGetUserNameFromId(UserId, &UserName) != 0) {
+
+        assert(UserName == NULL);
+
+        printf("%s", UserName);
```
Comment (ccstevens) [inline apps/swiss/whoami.c]: Might not want to print a NULL `UserName` here. :) I think you were intending to print the `UserId`?

File: apps/swiss/whoami.c [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+            Status = 1;
+            goto MainEnd;
+        }
+    }
+
+    Status = 0;
+    EffectiveUserId = SwGetEffectiveUserId();
+    WhoamiPrintUserName(EffectiveUserId);
+
```
Comment (ccstevens) [inline apps/swiss/whoami.c]: This blank line doesn't fit the Minoca style guide (not your fault, we haven't written it up).

We have a style checker that you can run out of our tools repository. We typically run it using [testpass.sh](https://gitlab.com/minoca/tools/blob/master/win32/scripts/testpass.sh), but you can also run it against just one architecture (easy to figure out from the bottom of that script).

File: apps/swiss/whoami.c [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    if (SwGetUserNameFromId(UserId, &UserName) != 0) {
+
+        assert(UserName == NULL);
+
+        printf("%s", UserName);
+        return;
+    }
+
+    printf("%s", UserName);
+
```
Comment (ccstevens) [inline apps/swiss/whoami.c]: This blank line also doesn't fit the style.

File: apps/swiss/whoami.c (hunk 1 above)
Comment (vascocosta) [inline apps/swiss/whoami.c]: Oh, that was a copy/paste mistake. :) UserId for sure.

**Comment by ccstevens** [comment]
Ah, and yes. We'll need you to sign the CA. Thanks.

File: apps/swiss/whoami.c (hunk 2 above)
Comment (vascocosta) [inline apps/swiss/whoami.c]: I always try to follow the style and was about to ask you guys where I could find some spec. Thanks for the pointer.

**Comment by vascocosta** [comment]
I just submitted the CA.

C:/Users/vasco/src/minoca/os/apps/swiss# testpass.sh
All RTL tests passed.
All MM tests passed.
All C library tests passed.
Scanning source style...

No violations found.

**Comment by ccstevens** [comment]
The style script is better than me. Totally missed that variable alphabetization issue. Looks good! I will work on getting this into our branch and then back up to github later today. (Fixing the irssi/ TCP connect issue now.)

**Comment by vascocosta** [comment]
As a seasoned kernel developer you're closer to an AI than I am. Still human, nevertheless! :)

I can't wait to see the diff from those TCP fixes.

**Comment by ccstevens** [comment]
I've merged this into our internal repository and it should get pushed out to Github soon, likely later today.

**Pull request closed without merging**